dlt/common/data_writers/buffered.py

```diff
@@ -79,10 +81,20 @@ def write_data_item(self, item: TDataItems, columns: TTableSchemaColumns) -> Non
         if isinstance(item, List):
             # items coming in single list will be written together, not matter how many are there
             self._buffered_items.extend(item)
+            # update row count, if item supports "num_rows" it will be used to count items
+            if len(item) > 0 and hasattr(item[0], "num_rows"):
+                self._buffered_items_count += reduce(lambda prev, tbl: prev + tbl.num_rows, item, 0)
```

Comment:
I think list comprehension is always faster: `sum([tbl.num_rows for tbl in item])`

Reply:
yes, it will be faster than using lambdas...